compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/SwingPanel.desktop.kt

```diff
@@ -299,3 +313,69 @@ private class Updater<T : Component>(
         isDisposed = true
     }
 }
+
+private fun Rect.round(density: Density): IntRect {
+    val left = floor(left / density.density).toInt()
+    val top = floor(top / density.density).toInt()
+    val right = ceil(right / density.density).toInt()
+    val bottom = ceil(bottom / density.density).toInt()
+    return IntRect(left, top, right, bottom)
+}
+
+private class InteropPointerInputModifier<T : Component>(
+    private val root: Container,
+    private val componentInfo: ComponentInfo<T>,
+) : PointerInputFilter(), PointerInputModifier {
+    override val pointerInputFilter: PointerInputFilter = this
+
+    override fun onPointerEvent(
+        pointerEvent: PointerEvent,
+        pass: PointerEventPass,
+        bounds: IntSize,
+    ) {
+        val dispatchDuringInitialTunnel = pointerEvent.changes.fastAny {
+            it.changedToDownIgnoreConsumed() || it.changedToUpIgnoreConsumed()
+        }
+        if (pass == PointerEventPass.Initial && dispatchDuringInitialTunnel) {
```

Comment:
Could you explain the logic here, and write a comment?

It is not obvious why we dispatch initial down/up (`dispatchDuringInitialTunnel == true`), and final move/enter/exit/scroll (`dispatchDuringInitialTunnel == false`)

Reply:
It's copied from Android's `PointerInteropFilter`. The explanation is here: https://github.com/JetBrains/compose-multiplatform-core/blob/976f6feab9f7539a02c264c8adc1a6b35eadecd1/compose/ui/ui/src/androidMain/kotlin/androidx/compose/ui/input/pointer/PointerInteropFilter.android.kt#L135-L141

Actually, it seems that we can simplify it by choosing any of the passes 🤔

Reply:
Could you add a comment here with a link to that code?

Reply:
Done